Library/Homebrew/download_strategy.rb

```diff
@@ -419,10 +419,12 @@ def fetch(timeout: nil)
         # Authorization is no longer valid after redirects
         meta[:headers]&.delete_if { |header| header.start_with?("Authorization") } if is_redirection
 
-        # The cached location is no longer fresh if Last-Modified is after the file's timestamp
-        if cached_location.exist? && last_modified && last_modified > cached_location.mtime
-          use_cached_location = false
-        end
+        # The cached location is no longer fresh if either:
+        # - Last-Modified value is newer than the file's timestamp
+        # - Content-Length value is different than the file's size
+        newer_last_modified = last_modified && last_modified > cached_location.mtime
+        different_file_size = file_size && file_size != cached_location.size
+        use_cached_location = false if cached_location.exist? && (newer_last_modified || different_file_size)
```

Comment:
This might be inscrutable, but if you want to short-circuit:
```suggestion
        use_cached_location &&= cached_location.exist? && (newer_last_modified || different_file_size)
```